From bdb923654c766a36c04259fe204e8203878befff Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Sun, 15 Mar 2020 23:07:09 -0700
Subject: [PATCH 1278/1852] v4l: xilinx: sdirxss: Add support for ST2081-10
 Mode 2 in 6G mode

commit 953dbf05f35b5102a5fffd4cd258cf3889e365f8 from
https://github.com/Xilinx/linux-xlnx.git

Add support for ST2081-10 Mode 2 which allows carriage of 1080 line
source image formats and ancillary data in a single link 6G-SDI
interface. These different image formats are RGB/YUV444/YUV422 10/12 bpc
in 2048/1920x1080p60/59.97/50/48/47.95 in 6G mode.

For details refer to ST2081-10:2018 Section 5 Table 6.
https://ieeexplore.ieee.org/stamp/stamp.jsp?tp=&arnumber=8320055

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-sdirxss.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-sdirxss.c b/drivers/media/platform/xilinx/xilinx-sdirxss.c
index beddfffe89f4..a560814158de 100644
--- a/drivers/media/platform/xilinx/xilinx-sdirxss.c
+++ b/drivers/media/platform/xilinx/xilinx-sdirxss.c
@@ -211,6 +211,7 @@
 #define XST352_BYTE1_ST372_2x720L_3GB		0x8B
 #define XST352_BYTE1_ST372_2x1080L_3GB		0x8C
 #define XST352_BYTE1_ST2081_10_2160L_6G		0xC0
+#define XST352_BYTE1_ST2081_10_2_1080L_6G	0xC1
 #define XST352_BYTE1_ST2081_10_DL_2160L_6G	0xC2
 #define XST352_BYTE1_ST2082_10_2160L_12G	0xCE
 
@@ -1153,6 +1154,13 @@ static int xsdirx_get_stream_properties(struct xsdirxss_state *state)
 			else
 				format->width = 3840;
 			break;
+		case XST352_BYTE1_ST2081_10_2_1080L_6G:
+			format->height = 1080;
+			if (active_luma)
+				format->width = 2048;
+			else
+				format->width = 1920;
+			break;
 		default:
 			dev_dbg(core->dev, "Unknown 6G Mode SMPTE standard\n");
 			return -EINVAL;
-- 
2.31.1

